### PR TITLE
 JDK-8276447 Deprecate finalization-related methods for removal

### PR DESCRIPTION
**Pull request by bchristi-git**
Here are the code changes for the "Deprecate finalizers in the standard Java API" portion of JEP 421 ("Deprecate Finalization for Removal") for code review.

This change makes the indicated deprecations, and updates the API spec for JEP 421. It also updates the relevant @SuppressWarning annotations.

The CSR has been approved.
An automated test build+test run passes cleanly (FWIW :D ).

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed
- [x] Change requires a CSR request to be approved

### Issues
 * [JDK-8276447](https://bugs.openjdk.java.net/browse/JDK-8276447): Deprecate finalization-related methods for removal
 * [JDK-8276748](https://bugs.openjdk.java.net/browse/JDK-8276748): Deprecate finalization-related methods for removal (**CSR**)

### Reviewers
 * [Roger Riggs](https://openjdk.java.net/census#rriggs) (@RogerRiggs - **Reviewer**) ⚠️ Review applies to eca95cb24efb42267c534cbc542eb9aa826b29e2
 * [Alan Bateman](https://openjdk.java.net/census#alanb) (@AlanBateman - **Reviewer**) ⚠️ Review applies to eca95cb24efb42267c534cbc542eb9aa826b29e2
 * [Lance Andersen](https://openjdk.java.net/census#lancea) (@LanceAndersen - **Reviewer**) ⚠️ Review applies to eca95cb24efb42267c534cbc542eb9aa826b29e2
 * [Joe Darcy](https://openjdk.java.net/census#darcy) (@jddarcy - **Reviewer**) ⚠️ Review applies to eca95cb24efb42267c534cbc542eb9aa826b29e2
 * [Mandy Chung](https://openjdk.java.net/census#mchung) (@mlchung - **Reviewer**) ⚠️ Review applies to eca95cb24efb42267c534cbc542eb9aa826b29e2
 * [Sergey Bylokhov](https://openjdk.java.net/census#serb) (@mrserb - **Reviewer**) ⚠️ Review applies to eca95cb24efb42267c534cbc542eb9aa826b29e2
 * [Stuart Marks](https://openjdk.java.net/census#smarks) (@stuart-marks - **Reviewer**) ⚠️ Review applies to 8cde06747394ac3f7c0ca2e6da93b3c32c2da2eb
 * [Phil Race](https://openjdk.java.net/census#prr) (@prrace - **Reviewer**) ⚠️ Review applies to 8cde06747394ac3f7c0ca2e6da93b3c32c2da2eb

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/6465/head:pull/6465` \
`$ git checkout pull/6465`

Update a local copy of the PR: \
`$ git checkout pull/6465` \
`$ git pull https://git.openjdk.java.net/jdk pull/6465/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 6465`

View PR using the GUI difftool: \
`$ git pr show -t 6465`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/6465.diff">https://git.openjdk.java.net/jdk/pull/6465.diff</a>

</details>

### CODE REVIEW

**Review by RogerRiggs — APPROVED** [review]
LGTM

**Review by AlanBateman — APPROVED** [review]
Looks good.

**Review by LanceAndersen — APPROVED** [review]

**Review by jddarcy — APPROVED** [review]

File: src/java.base/share/classes/java/lang/Object.java [hunk 1]
```diff
@@ -478,6 +478,12 @@ public final void wait(long timeoutMillis, int nanos) throws InterruptedExceptio
      * A subclass overrides the {@code finalize} method to dispose of
      * system resources or to perform other cleanup.
      * <p>
+     * <b>When running in a Java virtual machine in which finalization has been
```
Comment (mlchung) [inline src/java.base/share/classes/java/lang/Object.java:481]: Disabling finalization is not part of the Java SE spec.   This paragraph describes the implementation of HotSpot VM and I think it does not belong to this javadoc.

Comment (bchristi-git) [reply src/java.base/share/classes/java/lang/Object.java:481]: The coupling between this change and [8276422](https://bugs.openjdk.java.net/browse/JDK-8276422) ("Add command-line option to disable finalization") is probably tighter than would be ideal.  That [CSR](https://bugs.openjdk.java.net/browse/JDK-8276773) allows for finalization to be disabled in the SE Platform Spec and the JLS.

Comment (mlchung) [reply src/java.base/share/classes/java/lang/Object.java:481]: Thanks for pointing out that the CSR for JDK-8276422 ("Add command-line option to disable finalization") includes the change of the SE Platform Spec and JLS to allow an implementation for finalization to be disabled.   This makes senses now.

**Review by mlchung — APPROVED** [review]

**Review by mrserb — APPROVED** [review]

File: src/jdk.jconsole/share/classes/sun/tools/jconsole/SummaryTab.java [hunk 2]
```diff
@@ -113,6 +113,7 @@ protected void done() {
 
     StringBuilder buf;
 
+    @SuppressWarnings("deprecation")
```
Comment (stuart-marks) [inline src/jdk.jconsole/share/classes/sun/tools/jconsole/SummaryTab.java:116]: Item for future cleanup: refactor the call to `getObjectPendingFinalizationCount()` at line 219 (!) into a local variable declaration, and then move the warnings suppression to that declaration. This reduces the scope of warnings suppression.

**Review by stuart-marks — APPROVED** [review]

**Review by prrace — APPROVED** [review]

**Pull request closed without merging**